File: C++/1190. Reverse Substrings Between Each Pair of Parentheses.cpp
class Solution {
public:
    string reverseParentheses(string s) {
        vector<char> ss;
        for(auto c:s) {
            if(c != ')') {
                ss.push_back(c);
            } else {
                char t = ss.back();
                ss.pop_back();
                vector<char> temp;
                while(t != '(') {
                    temp.push_back(t);
                    t = ss.back();
                    ss.pop_back();
                }
                ss.insert(ss.end(), temp.begin(), temp.end());
            }
        }
        string res;
        for(auto c:ss) {
            res += c;
        }
        return res;
    }
};

File: Python3/1190. Reverse Substrings Between Each Pair of Parentheses.py
class Solution:
    def reverseParentheses(self, s: str) -> str:
        stack = []
        last = []

        for c in s:
            if c != ')':
                stack.append(c)
            else:
                temp = []
                t = stack.pop()
                while(t != '('):
                    temp.append(t)
                    t = stack.pop()
                stack += temp
        
        return ("").join(stack)
